packages/teleport/src/components/BannerList/useAlerts.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { useState, useEffect } from 'react';
+import Logger from 'shared/libs/logger';
+
+import { fetchClusterAlerts } from 'teleport/services/alerts';
+import useStickyClusterId from 'teleport/useStickyClusterId';
+
+import type { ClusterAlert } from 'teleport/services/alerts';
+
+const logger = Logger.create('ClusterAlerts');
+
+const DISABLED_BANNERS = 'disabledAlerts';
+
+function addHours(date: number, hours: number) {
+  return date + hours * 60 * 60 * 1000;
+}
+
+function getItem(key: string): string | null {
+  return window.localStorage.getItem(key);
+}
+
+function setItem(key: string, data: string) {
+  window.localStorage.setItem(key, data);
+}
+
+type DismissedAlert = {
+  [alertName: string]: string;
+};
+
+export function useAlerts() {
+  const [alerts, setAlerts] = useState<ClusterAlert[]>([]);
+  const [dismissedAlerts, setDismissedAlerts] = useState<DismissedAlert[]>([]);
+  const { clusterId } = useStickyClusterId();
+
+  useEffect(() => {
+    const disabledAlerts = getItem(DISABLED_BANNERS);
+    if (disabledAlerts) {
+      // Loop through the existing ones and remove those that have passed 24h.
+      const data = JSON.parse(disabledAlerts);
+      Object.entries(data).forEach(([name, expiry]) => {
+        if (new Date().getTime() > expiry) {
+          delete data[name];
+        }
+      });
+      setDismissedAlerts(data);
+      setItem(DISABLED_BANNERS, JSON.stringify(data));
+    }
+  }, []);
+
+  useEffect(() => {
+    fetchClusterAlerts(clusterId)
+      .then(res => {
+        if (!res) {
+          return;
+        }
+        const dismissedNames = Object.keys(dismissedAlerts);
+
+        const newAlerts = res.filter(alert => {
+          return !dismissedNames.includes(alert.metadata.name);
+        });
+        setAlerts(newAlerts);
+      })
+      .catch(err => {
+        logger.error(err);
+      });
+  }, [clusterId, dismissedAlerts]);
```

Comment:
i think the `dissmissedAlerts` dependency will trigger another alert fetch when we `setDismissedAlerts` from the useEffect above this one

Reply:
Yeah, I think fetching should be separated from filtering here.

```tsx
  useEffect(() => {
    fetchClusterAlerts(clusterId)
      .then(res => {
        if (!res) {
          return;
        }
        setAlerts(newAlerts);
      })
      .catch(err => {
        logger.error(err);
      });
  }, [clusterId]);
  
  const dismissedNames = Object.keys(dismissedAlerts);

  const visibleAlerts = alerts.filter(alert => !dismissedNames.includes(alert.metadata.name));
  
  return {
    alerts: visibleAlerts,
    dismissAlert,
  };
```

Reply:
Yep good catch! Fixed